File: unity/BlackjackUnity/Assets/Debugger.cs
using UnityEngine;
using System;
using System.Collections;

/// <summary>
/// Debugger helper class.
/// </summary>
public static class Debugger {

    /// <summary>
    /// Emit a Debug message.
    /// </summary>
    /// <param name="debug">Enable debug?</param>
    /// <param name="message">String to emit</param>
    public static void Emit(bool debug, string message) {
        if (debug) Debug.Log(message);
    }

    /// <summary>
    /// Emit a Debug message.
    /// </summary>
    /// <param name="debug">Enable debug?</param>
    /// <param name="integer">Integer to emit.</param>
    public static void Emit(bool debug, int integer) {
        Debugger.Emit(debug, integer.ToString());
    }

    /// <summary>
    /// Emit a Debug message.
    /// </summary>
    /// <param name="debug"></param>
    /// <param name="strings">string[] to emit.</param>
    public static void Emit(bool debug, string[] strings) {
        string msg = "";
        for (int i = 0; i < strings.Length; i++)
        {
            msg += String.Format("{0} ", strings[i]);
        }
        Debugger.Emit(debug, msg);
    }
}

File: java/io/stpettersen/blackjack/Debugger.java
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 * 
 * Java Swing/AWT implementation.
*/
package io.stpettersen.blackjack;

public class Debugger
{
    /**
     * Emit a debug message.
     * @param debug Enable debug?
     * @param message String to emit.
    */
    public static void emit(boolean debug, String message)
    {
        if(debug) System.out.println(message);
    }
    
    /**
     * Emit a debug message.
     * @param debug Enable debug?
     * @param integer Integer to emit.
    */
    public static void emit(boolean debug, int integer)
    {
        emit(debug, Integer.toString(integer));
    }
    
    /**
     * Emit a debug message.
     * @param debug Enabele debug?
     * @param e Exception to emit.
    */
    public static void emit(boolean debug, Exception e)
    {
        emit(debug, String.format("EXCEPTION: %s", e.getMessage()));
    }
}
